### PR TITLE
Enable Sabre for control flow circuits at O1

### LINKED ISSUE #9422: Restore level 1 layout/routing default to `SabreSwap`
### What should we add?

Part of #9417.  Depends on:
- #9419
- #9421

In #8830 we had to revert optimisation level 1 to use its legacy behaviour of laying out with `DenseLayout` and routing with `StochasticSwap` if the circuit was a control-flow circuit, since the new default `SabreSwap` couldn't handle the control-flow.  Once `SabreSwap` supports control-flow, we should remove this switching logic, and swap back to the preferred router.

`SabreLayout` should be easier to upgrade to initial support of control-flow (especially since it'll technically already run, it'll just not have good output), if only the Python-space paths are upgraded.

### PR DESCRIPTION
**Pull request by kevinhartman**
### Summary
Enables Sabre layout and routing for control flow circuits at optimization level 1.

### Details and comments
Depends on #10366.

Resolves #9422

### CODE REVIEW

File: releasenotes/notes/sabre-ctrl-flow-o1-431cd25a19adbcdc.yaml
Comment (mtreinish) [inline releasenotes/notes/sabre-ctrl-flow-o1-431cd25a19adbcdc.yaml:2]: This probably should be an upgrade note, since it's internal details for users of `transpile()` but it does mean that the layout and routing will produces different circuits when using control flow.

**Review by mtreinish — APPROVED** [review]
This LGTM, pending a rebase after #10366 merges

**Review by jakelishman — APPROVED** [review]

**Pull request merged**
